app/models/affiliate.rb

```diff
@@ -584,6 +592,14 @@ def set_managed_links(managed_links_attributes, managed_links)
     end
   end
 
+  def empty_link?(link)
+    empty_link = link['title'].blank? && link['url'].blank?
+    existing_link = link['id'].present?
+    # delete newly empty existing link
+    link[:_destroy] = 1 if empty_link && existing_link
```

Comment:
The issue with this is that the "destroying a link" logic gets buried into something completely unrelated,`#empty_link?` is now doing too much.

I think we could either have a hidden delete check box that gets checked when the link inputs get emptied. 

Or, in the backend, we can call `mark_for_destruction` on a link if title and link are blank, something like.

```ruby
class Link
  ...
  before_validate :mark_for_destruction, if: has_id_but_empty?

  def has_id_but_empty?
    !new_record? && (!title.present? && !url.present?)
  end
end
```

What do you think?

Reply:
Thanks @stevenbarragan .  Much cleaner.  Updates made.